### ISSUE TITLE
element order is adhered to even if ElementSelectors.byName is specified

### ISSUE THREAD

**Issue by ferng** [xmlunit/xmlunit#258]
adding this to Diff builder
`.withNodeMatcher(new DefaultNodeMatcher(ElementSelectors.byName))`

has no effect on the result if elements are out of order even in something like (well this exactly as  this is how i found it), i'm not using a dtd to validate against:
`            documentBuilderFactory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);`

sample xml

```
<outer>
  <middle data="anything" url="http://...." status="200">
    <inner name="Domain.field1" content="data1"/>
    <inner name="Domain.field2" content="data2"/>
  </middle>
...
</outer>
```

this seems to be because it always checks the sequence anyway
https://github.com/xmlunit/xmlunit/blob/main/xmlunit-core/src/main/java/org/xmlunit/diff/DOMDifferenceEngine.java#L611

**Comment by bodewig** [comment]
I'm not sure what you feel is wrong. `ElementSelector` tells XMLUnit which nodes to compare.

One of the comparisons is always the child node sequence. Whether this difference is important to you or not is not determined by the `ElementSelector` but by the `DifferenceEvaluator`, see https://github.com/xmlunit/user-guide/wiki/DifferenceEvaluator . The default `DifferenceEvaluator` will flag order differences as "SIMILAR", not "DIFFERENT". If even this is too much you can chain the default with an evaluator the downgrades the difference to "EQUAL", see https://github.com/xmlunit/xmlunit/blob/main/xmlunit-core/src/main/java/org/xmlunit/diff/DifferenceEvaluators.java#L139 .

**Comment by bodewig** [comment]
oh, and the DTD feature doesn't affect either `ElementSelector` nor `DifferenceEvaluator`.

**Comment by ferng** [comment]
Hi, thanks for the prompt reply, really appreciate it.

Funny you mentioned DifferenceEvaluator as I did end up using that as a way of downgrading the SIMILAR as to EQUAL (along with a few more checks to make sure it was OK), thank fully your code is really easy to read so it was easy to figure out.

I mentioned that I wasn't using the DTD as I knew it wasn't relevant and want to to clearly describe my scenario.

I only tried  ElementSelector after I saw your answer in stackOverflow:
https://stackoverflow.com/questions/33695041/compare-xml-ignoring-element-order

As my solution is correct I'll go ahead and close this issue
